src/app/games/lasca/lasca.component.ts

```diff
@@ -0,0 +1,330 @@
+        }
+    }
+    private rotateAdaptedBoardIfNeeded(): void {
+        if (this.role === Player.ONE) {
+            const rotatedAdaptedBoard: SpaceInfo[][] = [];
+            for (let y: number = 0; y < LascaState.SIZE; y++) {
+                rotatedAdaptedBoard[(LascaState.SIZE - 1) - y] = [];
+                for (let x: number = 0; x < LascaState.SIZE; x++) {
+                    rotatedAdaptedBoard[(LascaState.SIZE - 1) - y][(LascaState.SIZE - 1) - x] = this.adaptedBoard[y][x];
+                }
+            }
+            this.adaptedBoard = rotatedAdaptedBoard;
+        }
+    }
+    private hideLastMove(): void {
+        this.updateAdaptedBoardFrom(this.getState());
+        this.rotateAdaptedBoardIfNeeded();
+    }
+    public async onClick(x: number, y: number): Promise<MGPValidation> {
+        if (this.role === Player.ONE) {
+            x = (LascaState.SIZE - 1) - x;
+            y = (LascaState.SIZE - 1) - y;
+        }
+        const clickValidity: MGPValidation = this.canUserPlay('#space_' + x + '_' + y);
+        if (clickValidity.isFailure()) {
+            return this.cancelMove(clickValidity.getReason());
+        }
+        const clickedCoord: Coord = new Coord(x, y);
+        const clickedSpace: LascaSpace = this.getState().getPieceAt(clickedCoord);
+        const opponent: Player = this.getState().getCurrentOpponent();
+        if (clickedSpace.isCommandedBy(opponent)) {
+            return this.cancelMove(RulesFailure.CANNOT_CHOOSE_OPPONENT_PIECE());
+        }
+        if (this.validClicks.length === 0) {
+            return this.trySelectingPiece(clickedCoord);
+        } else {
+            return this.moveClick(clickedCoord);
+        }
+    }
+    public cancelMoveAttempt(): void {
+        this.validClicks = [];
+        this.capturedCoords = [];
+        this.updateAdaptedBoardFrom(this.getState());
+        this.rotateAdaptedBoardIfNeeded();
+    }
+    public async moveClick(clicked: Coord): Promise<MGPValidation> {
+        if (clicked.equals(this.validClicks[0])) {
+            this.cancelMoveAttempt();
+            return MGPValidation.SUCCESS;
+        }
+        const clickedSpace: LascaSpace = this.getState().getPieceAt(clicked);
+        const player: Player = this.getState().getCurrentPlayer();
+        if (clickedSpace.isCommandedBy(player)) {
+            this.cancelMoveAttempt();
+            return this.trySelectingPiece(clicked);
+        }
+        if (this.validClicks.length === 1) {
+            // Doing the second click, either a step or a first capture
+            const delta: Coord = this.validClicks[0].getVectorToward(clicked);
+            if (Math.abs(delta.x) === 1 && Math.abs(delta.y) === 1) {
+                // It is indeed a step
+                const step: LascaMove = LascaMove.fromStep(this.validClicks[0], clicked).get();
+                return this.chooseMove(step, this.getState());
```

Comment:
tiens, chooseMove prend un state, pourtant ce state est toujours this.getState() (ou une variable qui contient le même state que this.getState()). Tickettons la suppression de ce second argument, ou bien t'as une raison contre ?

Reply:
Ouais, ça peut être sympa ! Gedaan